primitives/src/assets.rs

```diff
@@ -71,9 +72,28 @@ pub trait AssetRegistrar<T: AssetConfig> {
 	fn update_asset_metadata(asset_id: AssetId, metadata: T::StorageMetadata) -> DispatchResult;
 }
 
-pub trait AssetConfig: 'static + Eq + Clone {
-	/// The trait we use to register Assets
-	type AssetRegistrar: AssetRegistrar<Self>;
+pub trait AssetConfig<C>: 'static + Eq + Clone
```

Comment:
Why does this require the implementer to be static and have a `Clone` and `Eq` implementation?

Reply:
probably not, I think I simply copied type definition of `frame_system::Config`.